### PR TITLE
dev_performanceimprov

### PR DESCRIPTION
**Pull request by JensBlack**
added crop functionality
added flatten multiple animal skeletons to simulate single animal dlc
added missing value handling

### CODE REVIEW

File: utils/generic.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def get_frames(self) -> tuple:
             if running_time <= 1 / FRAMERATE:
                 sleepy_time = int(np.ceil(1000/FRAMERATE - running_time / 1000))
                 cv2.waitKey(sleepy_time)
-        else:
+        elif REPEAT_VIDEO:
```
Comment (alexfrostdesu) [inline utils/generic.py:151]: We need to do something with case when video ends to not end it on crash. Maybe have some exception here and catch it in dlc.py?

**Review by alexfrostdesu — APPROVED** [review]
👍

File: utils/generic.py (hunk 1 above)
Comment (JensBlack) [reply utils/generic.py:151]: Not sure how to do this. Can you make a suggestion?

**Comment by JensBlack** [comment]
I've added a custom Exception and raised it in generic camera and video manager. catching it in the app for normal shutdown. Also i think I found a solution to restart the stream again by pressing stop stream/start stream. Can you check it out?

**Pull request merged**